### ISSUE TITLE
[Resolved] ClusterSuspace sorts supercell matrices in a wrong way

### ISSUE THREAD

**Issue by qchempku2017** [CederGroupHub/smol#261]
## Expected Behavior
in ClusterSubspace's supercell_orbit_mappings function, two symetrically inequivalent super-cell matrices (which means prim.make_supercell(matrix) are not identical by structure matcher) are expected to yield different orbit mappings, while two symmetrically equivalent structures are supposed to generate the same.

## Current Behavior

The current code is written as inherited from pyabinitio:
        scmatrix = np.array(scmatrix)
        # so change them into a tuple of sorted tuples for unique keys.
        scm = tuple(sorted(tuple(s.tolist()) for s in scmatrix))
        indices = self._supercell_orb_inds.get(scm)

        if indices is None:
            indices = self._gen_orbit_indices(scmatrix)
            self._supercell_orb_inds[scm] = indices
The current code considers supercell matrices as symmetrically equivalent after permutation in each row. This might be true with rhombohedral and cubic primitive cells, which we mostly study. But in any lattice system where a, b, c or alpha, beta, gamma are not equal, supercell matrices are apparently not equivalent after in-the-row permutations.

## Possible Solution

Change direct query of sorted(scmatrix) from self._supercell_orbit_inds dictionary, to checking symmetric equivalence of supercell matrix with the existing keys in the dictionary. This could be less efficient but more rigorous.

## Steps to Reproduce

1.
2.

**Comment by qchempku2017** [comment]
Another error found:
when using corr_from_structure to generate correlation functions of the same structure, but different scmatrix argument, although all scmatrix given are symmetrically equivalent, each will give a different correlation function, which is absolutely not
physical!
Test that fails:
`    for s in structures:
        npt.assert_array_almost_equal(cluster_subspace
                                      .corr_from_structure(s),
                                      cluster_subspace
                                      .corr_from_structure(s,
                                                           scmatrix=supercell_matrix))
        npt.assert_array_almost_equal(cluster_subspace
                                      .corr_from_structure(s),
                                      cluster_subspace
                                      .corr_from_structure(s,
                                                           scmatrix=matrix2))
        npt.assert_array_almost_equal(cluster_subspace
                                      .corr_from_structure(s),
                                      cluster_subspace
                                      .corr_from_structure(s,
                                                           scmatrix=matrix3))`

where structures are generated with:
`    structures = [gen_random_structure(cluster_subspace.structure,
                                       size=supercell_matrix)
                  for _ in range(10)]`
and matrix2, matrix 3 are:
`    matrix2 = (np.array([[0, -1, 0], [1, 0, 0], [0, 0, 1]])
               @ np.array(supercell_matrix, dtype=int))
    matrix3 = (np.array([[0, 0, 1], [-1, 0, 0], [0, -1, 0]])
               @ np.array(supercell_matrix, dtype=int))`

A reason for this is that, when generating corr_from_structure, our current code first calls occupancy_from_structure with scmatrix argument passed. Because of different site_mapping with different scmatrix, different occupancy strings are generated for each scmatrix. However, when corr_from_structure calls for orbit indices mapping, it gets from supercell_orbits_mapping, which gives the same indices mapping for all symetrically equivalent scmatrix, which apparently does not correspond to different site_mappings. 

In conclusion, the correct way of implementing orbit_inds_mapping is to generate one for each unique supercell matrix, and do not reduce based on any structural symmetry.

**Comment by qchempku2017** [comment]
It seems that with only row permutations, the current implementation can give the correct correlation functions. While with inversion or other Z(3) rotation manipulations, scmatrices will be hashed as different keys, so there would not be any issue. I was mistaken that I thought the original implementation sorted within rows. Case can be closed with more additional test added.
